kubetest2-gke/deployer/up.go

```diff
@@ -33,6 +35,17 @@ import (
 	"sigs.k8s.io/kubetest2/pkg/metadata"
 )
 
+// If one of the error patterns below is matched, it would be recommended to
+// retry creating the cluster in a different region.
+// - stockout
+// - nodes fail to start
+// - component is unhealthy
+var retryableCreationErrors = []*regexp.Regexp{
+	regexp.MustCompile(`.*does not have enough resources available to fulfill.*`),
+	regexp.MustCompile(`.*only \d+ nodes out of \d+ have registered; this is likely due to Nodes failing to start correctly.*`),
+	regexp.MustCompile(`.*All cluster resources were brought up, but: component .+ from endpoint .+ is unhealthy.*`),
```

Comment:
component unhealthy doesn't seem like a retryable error. we see similar issues many times for legitimate cluster bring up issues.

Reply:
Do you mean it won't succeed even if we retry in different regions/zones? And does it mean we'll have to allow the infra flakiness caused by this error?

Reply:
If it works after retrying in a different zone that just seems like a coincidence :) 
But that is very much a legitimate issue in most of our gke components testing, which shouldn't be retried.